titiler/pgstac/dependencies.py

```diff
@@ -26,13 +27,76 @@
 def SearchIdParams(
     search_id: Annotated[
         str,
-        Path(description="Search Id (pgSTAC Search Hash)"),
+        Path(description="PgSTAC Search Identifier (Hash)"),
     ]
 ) -> str:
     """search_id"""
     return search_id
 
 
+@cached(  # type: ignore
+    TTLCache(maxsize=cache_config.maxsize, ttl=cache_config.ttl),
+    key=lambda pool, collection_id: hashkey(collection_id),
+)
+@retry(
+    tries=retry_config.retry,
+    delay=retry_config.delay,
+    exceptions=(
+        pgErrors.OperationalError,
+        pgErrors.InterfaceError,
+    ),
+)
+def get_collection_id(pool: ConnectionPool, collection_id: str) -> str:
+    """Get Search Id for a Collection."""
+    search = model.PgSTACSearch(
```

Comment:
I would use the direct non cql parameter for collections here: collections=[collection_id]

Because we know that collection defined there is not possibly subject to any nested "OR" conditions, we do some additional optimizations on the pgstac side.

Reply:
🙏 I'll do. I'll create an example in the docs to show how this can of dependency could create complex `filter` search